Basic Mujoco setup and update `ros2_so_arm` submodule

sea-bass opened the pull request:
This PR adds a very basic MuJoCo - ros2_control setup with the SO-101 arm -- just wanted to get what I had out there, plus I added a statically-mounted camera to see how that worked.

```
ros2 launch pai_bringup so_arm_mujoco_bringup.launch.py
```

<img width="2489" height="1529" alt="image" src="https://github.com/user-attachments/assets/f50c0a32-e0c1-4f79-bef3-33f7fa8b4ad1" />


NOTE: I needed to make https://github.com/JafarAbdi/ros2_so_arm/pull/31 to configure the Gazebo example spawn location.

sea-bass committed:
Basic Mujoco setup

saikishor reviewed (commented):
Just these few things. Rest look great 🙌🏾 

Great job

saikishor reviewed (approved):
Looks spot on 👌🏽👌🏽

Yadunund reviewed (changes requested):
Thanks a lot @sea-bass for this amazing PR! It’s a great addition to the repo.

I have a few high-level suggestions to better align this with our existing architecture:
(The overall goal is to show reuse of the same assets and ROS nodes with different simulators.)

- Since `pai_mujoco` was developed independently, it currently duplicates some functionality. To keep the workspace clean, I recommend moving all configurations and launch files into the existing `pai_bringup` package and removing `pai_mujoco`. In the same or a follow up PR, we can split the logic in `so_arm_gz_bringup.launch.py` into separate launch files for starting the ROS controllers and another for bringing up the simulator. Then via launch args we can either start the gazebo sim or the mujoco one.
- The PR introduces two sources of truth for the robot description. We currently clone in a fork of https://github.com/JafarAbdi/ros2_so_arm100 which pulls in the URDF of the arm. But the [submodule](https://github.com/JafarAbdi/ros2_so_arm100/tree/main/so_arm100_description) in that repo is a bit outdated and does not include the new [mjcf](https://github.com/TheRobotStudio/SO-ARM100/tree/main/Simulation) files for the SO arms. We could bump the submodule version and rely on the upstream description. Note: I just opened https://github.com/JafarAbdi/ros2_so_arm100/pull/25 so that we can switch to the upstream repo instead of my fork.

sea-bass committed:
Merge remote-tracking branch 'origin/main' into scastro/mujoco

sea-bass committed:
Clean up

sea-bass committed:
Consolidate packages

sea-bass committed:
Fix and update SO-ARM submodule

Juliaj committed:
fix(mujoco): resolve ros2_control_node init and controller activation (#18)

- Add transmission_interface dependency for SO-ARM101 transmissions
- Add so_arm_mujoco_controllers.yaml with resolved joint names
- Sync scene.xml with external SO-ARM100 source of truth
- Add so-arm-mujoco Pixi task and MuJoCo docs

sea-bass committed:
Fixes

Juliaj commented on pai_bringup/scripts/lerobot_inference_node:
```diff
@@ -153,8 +153,8 @@ class LeRobotInferenceNode(Node):
-                'shoulder_pan_joint', 'shoulder_lift_joint', 'elbow_joint',
-                'wrist_pitch_joint', 'wrist_roll_joint', 'jaw_joint'
+                'shoulder_pan_joint', 'shoulder_lift_joint', 'elbow_flex_joint',
+                'wrist_pitch_joint', 'wrist_roll_joint', 'gripper_joint'
```
Is `wrist_pitch_joint` the same as the `wrist_flex_joint` used by the robot model (URDF, MJCF, LeRobot follower_arm.json etc) ? This joint name is also in pai_data_collection/config/rosetta/so_arm100.yaml under observations.

Juliaj replied:
```suggestion
                'wrist_flex_joint', 'wrist_roll_joint', 'gripper_joint'
```

Juliaj committed:
MuJoCo + rosetta data collection (#19)

sea-bass committed:
Use same controllers file in all cases

sea-bass commented on pai_bringup/config/ros2_controllers_mujoco.yaml:
```diff
@@ -0,0 +1,43 @@
+# MuJoCo-specific controller config for LeRobot inference and rosetta data collection.
```
For sure, fixed!

sea-bass committed:
No need for relay node

sea-bass commented on pai.repos:
```diff
@@ -1,15 +1,16 @@
     version: main
+  mujoco_ros2_control:
+    type: git
+    url: https://github.com/ros-controls/mujoco_ros2_control.git
```
Is this going to make it upstream? If so, I'd rather just wait for that to be available.

Juliaj reviewed (approved):
A few var renaming comments. Overall looks great, 🚀

sea-bass committed:
Apply suggestions from code review

Co-authored-by: Julia Jia <juliajster@gmail.com>

sea-bass committed:
missed one

Co-authored-by: Julia Jia <juliajster@gmail.com>

Yadunund reviewed (approved):
LGTM! Thanks for iterating
